### DISCUSSION TITLE
Issue Forms render form preview and validate yaml before commit

### DISCUSSION THREAD

**Discussion by leonard84** [community/community#4402] [Projects and Issues]
I really like the new form feature for the issues, but authoring the yaml files for that is unnecessarily cumbersome.
When you are editing a `<form-name>.yml` in the frontend, it already detects that you are editing a special issue form yaml and shows you some helpful documentation on the right hand side. However, if you click on the preview tab, you don't see a preview of the form, instead you just the the commit diff.
To actually see the form you have to submit it, and only then can you see the form and any yaml errors you made.
A pre-commit form preview and yaml validation would avoid useless commits, and make the whole experience much smoother.

**Comment by Kixunil** [comment]
Yeah, I had to do two fixup comits just when trying out the feature. Validation would be incredibly helpful.

**Comment by gavinr** [comment]
The diff isn't even showing for me:
![image](https://user-images.githubusercontent.com/209355/140996771-e3867f77-9c7b-498f-83c5-777332c39227.png)

Please support proper previews!

**Reply by Ridihush001** [reply]
Can't find where to state my issue

**Comment by melinath** [comment]
+1 - I can sort of get a preview after pushing the file to GitHub, but it doesn't actually look the same as the form looks after merge.
